packages/gestalt/src/BannerOverlay.js

```diff
@@ -0,0 +1,276 @@
+ * ![BannerOverlay light mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/BannerOverlay.spec.mjs-snapshots/BannerOverlay-chromium-darwin.png)
+ */
+export default function BannerOverlay({
+  message,
+  title,
+  dismissButton,
+  helperLink,
+  primaryAction,
+  offsetTop = 130,
+  offsetBottom = 24,
+  thumbnail,
+  zIndex,
+}: Props): ReactNode {
+  const { name: colorSchemeName } = useColorScheme();
+  const isDarkMode = colorSchemeName === 'darkMode';
+
+  const isMobileWidth = useDeviceType() === 'mobile';
+
+  let messageTextElement: Element<'span'> | string;
+
+  if (typeof message === 'string') {
+    messageTextElement = message;
+  }
+  if (typeof title === 'string') {
+    messageTextElement = title;
+  }
+
+  // If `message` is a Text component, we need to override any text colors within to ensure they all match
+  const messageIsTextNode =
+    typeof message !== 'string' &&
+    Children.only<Element<typeof Text>>(message).type.displayName === 'Text';
+
+  if (messageIsTextNode) {
+    const textColorOverrideStyles = isDarkMode
+      ? styles.textColorOverrideDark
+      : styles.textColorOverrideLight;
+
+    messageTextElement = <span className={textColorOverrideStyles}>{message}</span>;
+  }
+
+  const { accessibilityDismissButtonLabel: accessibilityDismissButtonLabelDefault } =
+    useDefaultLabelContext('BannerOverlay');
+
+  const { lightModeBackground, darkModeBackground, textColor } = DEFAULT_COLORS;
+
+  const dismissButtonComponent = dismissButton && (
+    <Flex.Item flex="none" alignSelf={isMobileWidth ? 'end' : 'center'}>
+      <IconButton
+        accessibilityLabel={
+          dismissButton.accessibilityLabel ?? accessibilityDismissButtonLabelDefault
+        }
+        icon="cancel"
+        iconColor="darkGray"
+        onClick={dismissButton.onDismiss}
+        size="xs"
+      />
+    </Flex.Item>
+  );
+  const messageComponent = (
+    <Flex.Item flex="grow">
+      <Text weight="bold">{title}</Text>
+      <BannerOverlayMessage
+        text={messageIsTextNode ? undefined : messageTextElement}
+        textElement={messageIsTextNode ? messageTextElement : undefined}
+        helperLink={helperLink}
+        textColor={textColor}
+      />
+    </Flex.Item>
+  );
+  const CTAComponent = primaryAction && (
+    <Flex.Item flex={isMobileWidth ? 'shrink' : 'none'}>
+      {primaryAction?.accessibilityLabel && primaryAction?.label
+        ? (primaryAction.role === 'link' && (
+            <PrimaryAction
+              accessibilityLabel={primaryAction.accessibilityLabel}
+              href={primaryAction.href}
+              label={primaryAction.label}
+              onClick={primaryAction.onClick}
+              rel={primaryAction?.rel}
+              role="link"
+              size="sm"
+              target={primaryAction?.target}
+            />
+          ),
```

Comment:
Im curious about this comma here... I've never done this myself before hahah

Reply:
could this work? 

```
primaryAction && primaryAction?.accessibilityLabel && primaryAction?.label (  <Flex.Item flex={isMobileWidth ? 'shrink' : 'none'}>
      {
        primaryAction.role === 'link'  ? (
            <PrimaryAction
              accessibilityLabel={primaryAction.accessibilityLabel}
              href={primaryAction.href}
              label={primaryAction.label}
              onClick={primaryAction.onClick}
              rel={primaryAction?.rel}
              role="link"
              size="sm"
              target={primaryAction?.target}
            />
          )
        : (  <PrimaryAction
              accessibilityLabel={primaryAction.accessibilityLabel}
              label={primaryAction.label}
              onClick={primaryAction.onClick}
              role="button"
              size="sm"
            /> ) }
    </Flex.Item>)

```

Reply:
Didn't understand this code syntax... Is this first line supposed to be a condition for a nested ternary?
Tried pasting it and vscode didn't understand it either, it look like a function signature heheh
The commas are the same behavior as an array, this has the same behavior and should be more familiar:
`const CTAComponent = primaryAction && (
    <Flex.Item flex={isMobileDevice ? 'shrink' : 'none'}>
      {primaryAction?.accessibilityLabel && primaryAction?.label
        ? [
            primaryAction.role === 'link' && (
              <PrimaryAction
                accessibilityLabel={primaryAction.accessibilityLabel}
                href={primaryAction.href}
                label={primaryAction.label}
                onClick={primaryAction.onClick}
                rel={primaryAction?.rel}
                role="link"
                size="sm"
                target={primaryAction?.target}
              />
            ),
            primaryAction.role !== 'link' && (
              <PrimaryAction
                accessibilityLabel={primaryAction.accessibilityLabel}
                label={primaryAction.label}
                onClick={primaryAction.onClick}
                role="button"
                size="sm"
              />
            ),
          ]
        : null}
    </Flex.Item>
  );`

Reply:
DONE